### DISCUSSION TITLE
Internal and External IPs

### DISCUSSION THREAD

**Discussion by darklord1807** [apache/doris#19567] [A - General / Q&A]
I am testing Doris but getting below issue: 

We have VMs that have internal and external IPs. 

eg. The internal IP is 192.168.0.2 and the external IP is 10.144.5.165

I have a FE and BE in the node. 

I have tried priority_networks = 10.144.5.0/24 but it fails to detect the IP and just goes to 127.0.0.1. When adding the BE, we get the error that 10.144.5.165 is not matching with 127.0.0.1. 

If i use priority network 192.168.0.0/24, BE works well and it works well. But when I try to streamload from external systems it looks for 192.168.0.2 and fails. 

Kindly advise. I have tried to use hostmode and FCQN. It still looks for 192.168.0.2.

**Comment by yagagagaga** [comment]
Maybe you can use internal IP in Local Area Network and use external IP in Wide Area Networks . There is no FCQN in Doris. Maybe you mean [FQDN](https://doris.apache.org/docs/dev/admin-manual/cluster-management/fqdn/). FQDN is a feature supported by Doris 2.0.

**Comment by darklord1807** [comment]
Apologies, FCQN. 

Is there anyway to hardcode/force or provide the BE IP that it should use.  Technically today it is trying to guess the IP itself.  It would never see the 10.144.5.165 which is an external IP.

**Comment by zy-kkk** [comment]
Hello, because the FE and BE of Doris are deployed separately and communicated through the internal network, but when you import data through Stream Load, you first connect to FE, at this time you can access through the external IP, and then FE will return the IP of BE, at this time, the internal IP will be returned. So you need to make sure your importer has access to the virtual machine's internal IP, or you can do a forward in the Stream Load program to convert the internal IP to the external IP

**Comment by darklord1807** [comment]
Is there anyway to force set the IP/override the IP of a BE node instead of it trying to get the IP?

**Comment by zy-kkk** [comment]
If you're just writing data directly to the Stream Load API, you can specify the BE ip:http_port
